test/e2e/storage/latebinding.go

```diff
@@ -0,0 +1,122 @@
+	Expect(pv.Spec.AccessModes).To(Equal(expectedAccessModes))
+	Expect(pv.Spec.ClaimRef.Name).To(Equal(claim.ObjectMeta.Name))
+	Expect(pv.Spec.ClaimRef.Namespace).To(Equal(claim.ObjectMeta.Namespace))
+	Expect(pv.Spec.PersistentVolumeReclaimPolicy).To(Equal(v1.PersistentVolumeReclaimDelete))
+
+	By(fmt.Sprintf("%s: deleting claim %q/%q", id, claim.Namespace, claim.Name))
+	framework.ExpectNoError(client.CoreV1().PersistentVolumeClaims(claim.Namespace).Delete(claim.Name, nil))
+
+	// Wait for the PV to get deleted if reclaim policy is Delete. (If it's
+	// Retain, there's no use waiting because the PV won't be auto-deleted and
+	// it's expected for the caller to do it.) Technically, the first few delete
+	// attempts may fail, as the volume is still attached to a node because
+	// kubelet is slowly cleaning up the previous pod, however it should succeed
+	// in a couple of minutes. Wait 20 minutes to recover from random cloud
+	// hiccups.
+	if pv.Spec.PersistentVolumeReclaimPolicy == v1.PersistentVolumeReclaimDelete {
+		By(fmt.Sprintf("%s: deleting the claim's PV %q", id, pv.Name))
+		framework.ExpectNoError(framework.WaitForPersistentVolumeDeleted(client, pv.Name, 5*time.Second, 20*time.Minute))
+	}
+}
+
+// PVWriteReadSingleNodeCheck checks that a PV retains data on a single node.
+//
+// It starts two pods:
+// - The first pod writes 'hello word' to the /mnt/test (= the volume) on one node.
+// - The second pod runs grep 'hello world' on /mnt/test on the same node.
+//
+// The node is selected by Kubernetes when scheduling the first
+// pod. It's then selected via its name for the second pod.
+//
+// If both succeed, Kubernetes actually allocated something that is
+// persistent across pods.
+//
+// This is a common test that can be called from a StorageClassTest.PvCheck.
+func PVWriteReadSingleNodeCheck(client clientset.Interface, claim *v1.PersistentVolumeClaim, id string) {
+	By(fmt.Sprintf("%s: checking the created volume is writable", id))
+	command := "echo 'hello world' > /mnt/test/data || (mount | grep 'on /mnt/test'; false)"
+	pod := testsuites.StartInPodWithVolume(client, claim.Namespace, claim.Name, "pvc-volume-tester-writer-"+id, command, testsuites.NodeSelection{})
+	defer func() {
+		// pod might be nil now.
+		testsuites.StopPod(client, pod)
+	}()
+	framework.ExpectNoError(framework.WaitForPodSuccessInNamespaceSlow(client, pod.Name, pod.Namespace))
+	runningPod, err := client.CoreV1().Pods(pod.Namespace).Get(pod.Name, metav1.GetOptions{})
+	Expect(err).NotTo(HaveOccurred(), "get pod %s", id)
+	actualNodeName := runningPod.Spec.NodeName
+	testsuites.StopPod(client, pod)
+	pod = nil // Don't stop twice.
+
+	By(fmt.Sprintf("%s: checking the created volume is readable and retains data on the same node %q", id, actualNodeName))
+	command = "grep 'hello world' /mnt/test/data"
+	testsuites.RunInPodWithVolume(client, claim.Namespace, claim.Name, "pvc-volume-tester-reader-"+id, command, testsuites.NodeSelection{Name: actualNodeName})
```

Comment:
Not familiar with the test frameworks used in e2e test, but shouldn't we check that the `command` was successfully executed? perhaps something like? 
* `framework.ExpectNoError()` call or
* `Expect(err).NotTo(HaveOccurred(), ,)` call

Reply:
There's no error to be checked here. The error handling is all inside `testsuites.RunInPodWithVolume`.

Reply:
Got it.